### PR TITLE
Carbon tracking - final work

### LINKED ISSUE #467: Tracking upgrades
As I finalize #466 , some thoughts:

- [x] We'll want a tracking vignette
- [x] Remove the diff stuff in the tracking code (after checking the need for it has been addressed)
- [x] Ideally the user can use a new function `fetchtracking()` to get the tracking output data from within R
- [x] Right now the tracking output file looks like this; rethink at all?

year | pool_name | pool_value | pool_units | source_name | source_fraction
-- | -- | -- | -- | -- | --
1746 | Diff | -0.6125 | U_PGC | veg_diff | 0
1746 | Diff | 15.7918 | U_PGC | soil_diff | 0

### PR DESCRIPTION
**Pull request by bpbond**
* Spent a while investigating the big diff values for atmosphere, detritus, and soil that occur, and I'm satisfied that this is a function of a small pool with high turnover; we have the solver values to adjust to, and that seems fine. Removing all the 'diff' related temporary code.
* Added `component` as a column in the tracking data output
* Expanded and clarified some documentation points

Closes #467

### CODE REVIEW

**Comment by bpbond** [comment]
@leeyap Could you review the changes to your vignette, as well as the `get_tracking_data()` documentation?

@kdorheim Ready for your review.

Thanks!

**Review by leeyap — APPROVED** [review]
Looks good to me. Thanks, @bpbond!

File: DESCRIPTION
Comment (kdorheim) [inline DESCRIPTION:3]: EPPPP!!! 🥳

File: src/simpleNbox-runtime.cpp
Comment (kdorheim) [inline src/simpleNbox-runtime.cpp:247]: When are these TODOs going to be addressed? Is there a github issue related to them?

File: DESCRIPTION
Comment (kdorheim) [inline DESCRIPTION:28]: Add Skylar!

File: vignettes/carbon_tracking.Rmd
Comment (kdorheim) [inline vignettes/carbon_tracking.Rmd:87]: Isn't 'component' also a name returned from the model output?

**Review by kdorheim — CHANGES_REQUESTED** [review]
It is really coming together! So exciting! A few minor comments & a few questions that I occurred to me while reviewing this PR, I don't necessarily know if they even need to be addressed or if this PR is the appropriate place to do so. 

1) setting the tracking var date tells us when we start tracking the carbon, how do users know what date is used in the final comparison? When fetch_tracking is called is it querying the core's current date? Should that information be returned with the data frame? 
2) is there a way to launch the tracking from the ini file? thinking about the cpp users does it return a csv file to the output dir?

File: src/simpleNbox-runtime.cpp
Comment (bpbond) [reply src/simpleNbox-runtime.cpp:247]: I don't think we have a GitHub issue but it'll be related to dealing with #397 . I will add as an issue too though.

File: DESCRIPTION
Comment (bpbond) [reply DESCRIPTION:28]: Doh! Thanks

Comment (bpbond) [reply DESCRIPTION:28]: @skygering Do you have an ORCID ID?

File: vignettes/carbon_tracking.Rmd
Comment (bpbond) [reply vignettes/carbon_tracking.Rmd:87]: Yes; I'm just talking about the three that overlap with the model output but differ in their names.

Comment (bpbond) [reply vignettes/carbon_tracking.Rmd:87]: Clarifying

**Comment by bpbond** [comment]
@kdorheim Thanks for the review!

>How do users know what date is used in the final comparison?

The first date in the tracking output will be the same as `trackingDate`. I'll add a note to the vignette clarifying this.

>is there a way to launch the tracking from the ini file?

Yes, it's a core-level parameter there. Adding a note clarifying this.

File: DESCRIPTION
Comment (bpbond) [reply DESCRIPTION:28]: Moving this to an issue

**Comment by bpbond** [comment]
@kdorheim Great comments—thank you. I think this is ready to merge once checks pass.

**Comment by bpbond** [comment]
...but of course will wait for your approval 😄

**Review by kdorheim — APPROVED** [review]

**Pull request merged**
